Grammar/python.gram

```diff
@@ -692,6 +692,8 @@ invalid_primary:
 invalid_comprehension:
     | ('[' | '(' | '{') a=starred_expression for_if_clauses {
         RAISE_SYNTAX_ERROR_KNOWN_LOCATION(a, "iterable unpacking cannot be used in comprehension") }
+    | ('[' | '{') a=star_named_expression ',' [star_named_expressions] {
```

Comment:
The '(' is not included to avoid confusion when using generator expressions in function calls. We want:

```
❯ ./python -c 'f(x,z for x,y in something)'
  File "<string>", line 1
    f(x,z for x,y in something)
        ^
SyntaxError: Generator expression must be parenthesized
```

not

```
❯ ./python -c 'f(x,z for x,y in something)'
  File "<string>", line 1
    f(x,z for x,y in something)
        ^
SyntaxError: did you forgot parentheses around the comprehension target?
```